### ISSUE TITLE
Error sending mail

### ISSUE THREAD

**Issue by rolandog** [robamler/dontprint#40]
Operation: sending

It might have been a very large PDF. The original file was a 600 page, 8 MB PDF that turns into a 120 MB file with k2pdfopt.

**Comment by rolandog** [comment]
Got the same issue with a 1.4 MB, 313 page PDF. Forgot to mention I'm using a Kindle PaperWhite (third gen).

After some research, I found the following information about [Amazon's file size limit](https://www.amazon.com/gp/help/customer/forums/kindleqna/ref=cs_hc_k_pl?ie=UTF8&forumID=Fx1FI6JDSFEQQ7V&cdThread=Tx12L4PEVSFX47U&cdPage=1&cdMsgId=Mx1P5ES546RGS8Q#Mx1P5ES546RGS8Q). It seems that sending emails with 50 MB or larger attachments isn't allowed.

Is there a way to try to minimize file sizes? Or perhaps some heuristics to try to warn users if their PDFs might result in a very large file?

**Comment by CsatiZoltan** [comment]
@rolandog  

> Is there a way to try to minimize file sizes?

Yes, there is. Pass the `-n` flag to k2pdfopt. It will keep the vector description of the text. Otherwise, it is converted to image (without OCR), that's why you got such a large file size. See http://www.willus.com/k2pdfopt/help/native.shtml
